### PR TITLE
Fix msys builds by downgrade

### LINKED ISSUE #6275: msys linking failure: libSDL2-image IMG-avif missing?
**Describe the bug**
mingw build failure in the workflow.

**To reproduce**
See https://github.com/widelands/widelands/actions/runs/7279661441/job/19836541027?pr=6150

**Build log**
```
[2352/2355] Linking CXX executable src\ai\test\test_ai.exe
FAILED: src/ai/test/test_ai.exe 
C:\Windows\system32\cmd.exe /C "cd . && C:\msys64\mingw32\bin\g++.exe -O3 -DNDEBUG -Wl,--large-address-aware @CMakeFiles\test_ai.rsp -o src\ai\test\test_ai.exe -Wl,--out-implib,src\ai\test\libtest_ai.dll.a -Wl,--major-image-version,0,--minor-image-version,0  && cd ."
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0xdd): undefined reference to `_imp__avifDecoderSetIO'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0xe5): undefined reference to `_imp__avifDecoderDestroy'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0xed): undefined reference to `_imp__avifDecoderCreate'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0xf9): undefined reference to `_imp__avifPeekCompatibleFileType'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0x105): undefined reference to `_imp__avifDecoderNextImage'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0x111): undefined reference to `_imp__avifImageYUVToRGB'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0x120): undefined reference to `_imp__avifResultToString'
C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/13.2.0/../../../../i686-w64-mingw32/bin/ld.exe: C:/msys64/mingw32/lib/libSDL2_image.a(libSDL2_image_la-IMG_avif.o):(.text+0x12c): undefined reference to `_imp__avifDecoderParse'

collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Version:**
 - OS: w*ndows workflow runner images
 - Widelands Version: https://github.com/widelands/widelands/pull/6150/commits/05bc2caeddcf5fc2026ab593811a3070b5025648 / https://github.com/tothxa/widelands/commit/05bc2caeddcf5fc2026ab593811a3070b5025648 (only data changes compared to current master / [9efa63b](https://github.com/widelands/widelands/commit/9efa63b84337274af12244ebecb58a17195e6c28))

**Additional context:**
We should consider building libSDL2 ourselves to avoid the constant dependency creep in msys.

### PR DESCRIPTION
**Pull request by hessenfarmer**
downgrade sdl2 image as no static lib is provided for new dependency Avif
fixes #6275

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -328,7 +330,7 @@ jobs:
         pacman.exe --noconfirm -Syyuu --ignore ca-certificates
         pacman.exe --noconfirm -Syyuu --ignore ca-certificates
         bash.exe -c "./install-dependencies.sh msys${{ matrix.bits }} --noconfirm"
-        # pacman.exe --noconfirm -U https://repo.msys2.org/mingw/$env:archname/mingw-w64-$env:archname-libtiff-4.3.0-3-any.pkg.tar.zst
+        pacman.exe --noconfirm -U https://repo.msys2.org/mingw${{ matrix.bits }}/mingw-w64-${{ matrix.archname }}-SDL2_image-2.8.0-1-any.pkg.tar.zst
```
Comment (matthiakl) [inline .github/workflows/build.yaml]:
```suggestion
        pacman.exe --noconfirm -U https://repo.msys2.org/mingw/mingw${{ matrix.bits }}/mingw-w64-${{ matrix.archname }}-SDL2_image-2.8.0-1-any.pkg.tar.zst
```

**Comment by hessenfarmer** [comment]
thanks for the patch. I'd never have realized this silly mistake.

BTW: 
A PR is already up to fix the issue in the LIBAVIF package of MSYS.  
see
https://github.com/msys2/MINGW-packages/pull/19512

**Review by matthiakl — APPROVED** [review]

**Comment by matthiakl** [comment]
@bunnybot merge

**Comment by Biswa96** [comment]
Please wait while this package is being fixed in msys2 repository.

**Comment by hessenfarmer** [comment]
@bunnybot update

**Comment by hessenfarmer** [comment]
> Please wait while this package is being fixed in msys2 repository.

as we need our Ci up and running as we are close to a release, waiting is no option. We had this in the past and we will disable the downgrade as soon as a solution in msys is available.

**Comment by hessenfarmer** [comment]
@bunnybot merge

**Pull request closed without merging**